### PR TITLE
Pb combustion chamber ref moist air

### LINKED ISSUE #506: [FEATURE] Combustion chamber with RefMoistAir as an inlet
### What is your precise need ?
To able to use `RefMoistAir` medium up until the combustion chamber without converting to flue gas before.
### What solution do you see ?
Create a new combustion chamber that takes `RefMoistAir` medium as an inlet from the compressor.
### Do you see any other alternative ?
The current alternative is to use the `.RefMoistAir_to_FlueGases` converter before entering the combustion chamber

### PR DESCRIPTION
**Pull request by phelimb-met**
## Goal

To able to use RefMoistAir medium up until the combustion chamber without converting to flue gas before.

Resolves #506 

## Type of change 

- [ ] Bugfix
- [x] New feature
- [ ] Refactoring change
- [ ] Release & Version Update (don't forget to change the version number in `package.mo`)

Will it break anything in previous models ?

- [ ] Breaking change (If yes, make sure to point it out in the changelog)
- [x] Non-Breaking change

## Checklist

- [x] I have added the appropriate tags, reviewers, projects (and detailed the size and priority of my PR) and linked issues to this PR
- [x] I have performed a self-review of my own code
- [x] I have checked that all existing tests pass
- [ ] I have checked that my work is compatible with [OpenModelica](https://openmodelica.org/)
- [x] I have added/updated tests that prove my development works and does not break anything.
- [x] I have made corresponding changes or additions to the documentation (in [Notion documentation](https://www.notion.so/metroscope/Metroscope-Modeling-Library-Documentation-MML3-WIP-50c8703c294446059d3b4a70d6ae4a71))
- [x] I have added corresponding entries to the [Changelog](../CHANGELOG.md)
- [x] I have checked for conflicts with target branch, and merged/rebased in consequence

You can also fill these out after creating the PR, but make sure to **check them all before submitting your PR for review**.

### CODE REVIEW

File: MetroscopeModelingLibrary/wslpath.bat [hunk 1]
```diff
@@ -0,0 +1 @@
+set WSLPath=C:/Windows/System32/wsl.exe
```
Comment (nabily-met) [inline MetroscopeModelingLibrary/wslpath.bat:1]: Delete this file

File: MetroscopeModelingLibrary/wslenv.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+DYMOLA="C:/Program Files/Dymola 2024x"
```
Comment (nabily-met) [inline MetroscopeModelingLibrary/wslenv.sh:1]: delete this file

File: MetroscopeModelingLibrary/call_wsl.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+IFS_SAVED=$IFS
+IFS=';'
+for p in $DYMOLALIB
+do
+  p=$(wslpath "$p");
+  LD_LIBRARY_PATH=$LD_LIBRARY_PATH":$p"
+done
+IFS=$IFS_SAVED
+
+"./$prog" "$@"
```
Comment (nabily-met) [inline MetroscopeModelingLibrary/call_wsl.sh:43]: Delete this file

File: MetroscopeModelingLibrary/call_wsl.bat [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+call "%WSLPath%" -e ./call_wsl.sh %*
+
+goto done
+
+:wslCorrect
+echo WSL not found: "%WSLPath%" >> %log%
+echo Please correct the WSL path. >> %log%
+echo For instructions on how to install WSL, please visit http://www.dymola.com/compiler >> %log%
+
+:done
```
Comment (nabily-met) [inline MetroscopeModelingLibrary/call_wsl.bat:24]: Delete this file

File: MetroscopeModelingLibrary/_wslbuild.sh [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+for p in $DYMOLALIB
+do
+  p=$(wslpath "$p");
+  LOPTS=$LOPTS" -L '$p'"
+done
+
+IFS=$IFS_SAVED
+
+# invoke dsbuild.sh
+"$DYMOLA/bin/dsbuild.sh" "$IOPTS" "$LOPTS" "$@"
```
Comment (nabily-met) [inline MetroscopeModelingLibrary/_wslbuild.sh:81]: Delete this file

File: MetroscopeModelingLibrary/MultiFluid/Machines/CombustionChamberWithRefMoistAir.mo [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  Wth = eta * Q_fuel * LHV;
+  Q_exhaust * h_exhaust = Q_air * h_in_air + Q_fuel * h_in_fuel + Wth;
+
+  // Chemical balance for fuel
+  X_fuel_C = m_C * (X_fuel_CH4/m_CH4 + 2*X_fuel_C2H6/m_C2H6 + 3*X_fuel_C3H8/m_C3H8 + 4*X_fuel_C4H10_n_butane/m_C4H10 + X_fuel_CO2/m_CO2);
+  X_fuel_H = m_H * (4*X_fuel_CH4/m_CH4 + 6*X_fuel_C2H6/m_C2H6 + 8*X_fuel_C3H8/m_C3H8 + 10*X_fuel_C4H10_n_butane/m_C4H10);
+  X_fuel_O = 2 * m_O * X_fuel_CO2/m_CO2;
+
+  // Mass balance for all species
+  - Q_exhaust * X_out_N2  + Q_air * 0.78  + Q_fuel * X_fuel_N2 = 0; // N2 balance
```
Comment (nabily-met) [inline MetroscopeModelingLibrary/MultiFluid/Machines/CombustionChamberWithRefMoistAir.mo:110]:
```suggestion
  - Q_exhaust*X_out_N2 + Q_air*0.78*(1 - sink_air.Xi_in[1]) + Q_fuel*X_fuel_N2 = 0; // N2 balance
```

File: MetroscopeModelingLibrary/MultiFluid/Machines/CombustionChamberWithRefMoistAir.mo [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+  - Q_exhaust * X_out_O2  + Q_air * 0.21  + Q_fuel * X_fuel_O * 0.5 = Q_fuel * m_O * (2 * X_fuel_C/m_C + 0.5 * X_fuel_H/m_H); // O2 balance
```
Comment (nabily-met) [inline MetroscopeModelingLibrary/MultiFluid/Machines/CombustionChamberWithRefMoistAir.mo:111]:
```suggestion
  - Q_exhaust*X_out_O2 + Q_air*0.22*(1 - sink_air.Xi_in[1]) + Q_fuel*X_fuel_O*0.5 = Q_fuel*m_O*(2*X_fuel_C/m_C + 0.5*X_fuel_H/m_H); // O2 balance
```

File: MetroscopeModelingLibrary/MultiFluid/Machines/CombustionChamberWithRefMoistAir.mo [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+  - Q_exhaust * X_out_H2O + Q_air * 0.01 = -Q_fuel * (0.5 * X_fuel_H/m_H) * m_H2O; // H2O balance
```
Comment (nabily-met) [inline MetroscopeModelingLibrary/MultiFluid/Machines/CombustionChamberWithRefMoistAir.mo:112]:
```suggestion
  - Q_exhaust*X_out_H2O + Q_air*sink_air.Xi_in[1] = - Q_fuel*(0.5*X_fuel_H/m_H)*m_H2O; // H2O balance
```

File: MetroscopeModelingLibrary/MultiFluid/Machines/CombustionChamberWithRefMoistAir.mo [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+  Q_exhaust = Q_air + Q_fuel;
+
+  // Mechanical balance
+  sink_air.P_in - source_exhaust.P_out = 0;
+  pressure_loss.delta_z = 0;
+  pressure_loss.Kfr = Kfr;
+
+  // Energy balance
+  Wth = eta * Q_fuel * LHV;
+  Q_exhaust * h_exhaust = Q_air * h_in_air + Q_fuel * h_in_fuel + Wth;
```
Comment (nabily-met) [inline MetroscopeModelingLibrary/MultiFluid/Machines/CombustionChamberWithRefMoistAir.mo:102]: Check enthalpies references

**Comment by hadrienp-met** [comment]
@phelimb-met is it finished ?

**Comment by phelimb-met** [comment]
> @phelimb-met is it finished ?

Yes

**Comment by hadrienp-met** [comment]
> > @phelimb-met is it finished ?
> 
> Yes

Great, you have to resolve the conversations and you can reask Nabil's review for final aprouval

**Review by nabily-met — APPROVED** [review]
Good job

**Pull request merged**